lib/streamlit/runtime/state/query_params.py

```diff
@@ -62,6 +62,10 @@ def __getitem__(self, key: str) -> str:
             raise KeyError(missing_key_error_message(key))
 
     def __setitem__(self, key: str, value: str | Iterable[str]) -> None:
+        self.__set_item_internal(key, value)
```

Comment:
Have you seen this method:
`set_with_no_forward_msg`? I think this should be using this method.